lib/doorkeeper/oauth/base_request.rb

```diff
@@ -26,16 +26,20 @@ def scopes
         @scopes ||= build_scopes
       end
 
-      def find_or_create_access_token(client, resource_owner, scopes, server)
+      def find_or_create_access_token(client, resource_owner, scopes, custom_fields, server)
         context = Authorization::Token.build_context(client, grant_type, scopes, resource_owner)
+        token_model = server_config.access_token_model
         application = client.is_a?(Doorkeeper::Application) ? client : client.application if client
-        @access_token = server_config.access_token_model.find_or_create_for(
+
+        token_params = {
```

Comment:
Let's name it as `token_attributes` and rename `custom_fields` into `custom_attributes`